### PR TITLE
feat: Migrate getByParamsUnformatted -> spenderPlatformV1ApiService

### PR DESCRIPTION
**Pull request by SahilK-027**
### Description

Migrate getByParamsUnformatted -> spenderPlatformV1ApiService
Screenshot 2024-05-08 at 4 39 56 PM

## Clickup
[Please add link here](https://app.clickup.com/t/86cv9n9ma)

### CODE REVIEW

**Review by Chethan-Fyle — APPROVED** [review]

**Review by devendrafyle — APPROVED** [review]

File: src/app/core/mock-data/platform/v1/platform-project.data.ts [hunk 1]
```diff
@@ -21,3 +22,47 @@ export const platformProjectSingleRes: PlatformApiResponse<PlatformProject[]> =
...
+      org_id: 'orFdTTTNcyye',
+      updated_at: new Date('2021-07-08T10:28:27.686886'),
+      name: 'Fyle Engineering',
+      sub_project: null,
+    },
+  ],
+  offset: 0,
+});
+
+export const ProjectPlatformParams: PlatformProjectParams = deepFreeze({
```
Comment (Dimple16) [inline src/app/core/mock-data/platform/v1/platform-project.data.ts:59]: this should be in a separate file `platform-project-params.data.ts`

File: src/app/core/models/platform/v1/platform-project-params.model.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export interface PlatformProjectParams {
+  limit: number;
+  offset: number;
+  order?: string;
+  sortDirection?: string;
+  sortOrder?: string;
+  searchNameText?: string;
+  active?: boolean;
```
Comment (Dimple16) [inline src/app/core/models/platform/v1/platform-project-params.model.ts:8]: why is active here?

File: src/app/core/models/platform/v1/platform-project-params.model.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  is_enabled?: string;
+  id?: string;
+  category_ids?: string;
```
Comment (Dimple16) [inline src/app/core/models/platform/v1/platform-project-params.model.ts:11]: why are there 2 category IDs? `category_ids` and `orgCategoryIds`

File: src/app/core/models/platform/v1/platform-project-params.model.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  org_id?: string;
+  name?: string;
+  orgCategoryIds?: string[];
```
Comment (Dimple16) [inline src/app/core/models/platform/v1/platform-project-params.model.ts:14]: I don't think this and the below `projectIds` is required. Can you check?

File: src/app/core/services/projects.service.spec.ts [hunk 5]
```diff
@@ -97,24 +101,27 @@ describe('ProjectsService', () => {
   });
 
   it('should be able to fetch data when no params provided', (done) => {
-    apiV2Service.get.and.returnValue(of(apiV2ResponseMultiple));
+    spenderPlatformV1ApiService.get.and.returnValue(of(platformAPIResponseMultiple));
 
     projectsService.getByParamsUnformatted({}).subscribe((res) => {
-      expect(res).toEqual(fixDate(apiV2ResponseMultiple.data));
```
Comment (Dimple16) [inline src/app/core/services/projects.service.spec.ts:103]: do we not require `fixDate` anymore? what has changed in the new response?

File: src/app/shared/components/fy-select-project/fy-select-modal/fy-select-project-modal.component.spec.ts [hunk 6]
```diff
@@ -277,24 +275,24 @@ describe('FyProjectSelectModalComponent', () => {
     });
   });
 
-  it('onDoneClick(): should dimiss the modal on clicking the done CTA', () => {
-    modalController.dismiss.and.returnValue(Promise.resolve(true));
+  it('onDoneClick(): should dimiss the modal on clicking the done CTA', async () => {
```
Comment (Dimple16) [inline src/app/shared/components/fy-select-project/fy-select-modal/fy-select-project-modal.component.spec.ts:278]: check if `fakeAsync` works in this case

File: src/app/core/services/recently-used-items.service.ts [hunk 7]
```diff
@@ -32,10 +32,10 @@ export class RecentlyUsedItemsService {
     ) {
       return this.projectsService
```
Comment (Dimple16) [inline src/app/core/services/recently-used-items.service.ts:33]: can you check if any of this is used in the expense service?

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/core/services/recently-used-items.service.ts (hunk 7 above)
Comment (SahilK-027) [reply src/app/core/services/recently-used-items.service.ts:33]: We have used it only in 
add-edit-expense
add-edit-advance
add-edit-per-deim
add-edit-mileage
fy-select-project-modal
utility-service
virtual-select-modal

File: src/app/core/services/projects.service.spec.ts (hunk 5 above)
Comment (SahilK-027) [reply src/app/core/services/projects.service.spec.ts:103]: 👍

File: src/app/core/models/platform/v1/platform-project-params.model.ts (hunk 2 above)
Comment (SahilK-027) [reply src/app/core/models/platform/v1/platform-project-params.model.ts:8]: DONE

File: src/app/core/mock-data/platform/v1/platform-project.data.ts (hunk 1 above)
Comment (SahilK-027) [reply src/app/core/mock-data/platform/v1/platform-project.data.ts:59]: DONE

File: src/app/core/models/platform/v1/platform-project-params.model.ts (hunk 3 above)
Comment (SahilK-027) [reply src/app/core/models/platform/v1/platform-project-params.model.ts:11]: DONE

File: src/app/core/models/platform/v1/platform-project-params.model.ts (hunk 4 above)
Comment (SahilK-027) [reply src/app/core/models/platform/v1/platform-project-params.model.ts:14]: DONE

File: src/app/shared/components/fy-select-project/fy-select-modal/fy-select-project-modal.component.spec.ts (hunk 6 above)
Comment (SahilK-027) [reply src/app/shared/components/fy-select-project/fy-select-modal/fy-select-project-modal.component.spec.ts:278]: I will do that with next PR for better test cases

File: src/app/core/services/projects.service.ts [hunk 8]
```diff
@@ -26,8 +27,8 @@ export class ProjectsService {
   @Cacheable()
   getByParamsUnformatted(
     projectParams: Partial<{
-      orgId: string;
-      active: boolean;
+      org_id: string;
+      is_enabled: boolean;
```
Comment (Dimple16) [inline src/app/core/services/projects.service.ts:31]: this should be camelCase

File: src/app/core/services/projects.service.ts [hunk 9]
```diff
@@ -26,8 +27,8 @@ export class ProjectsService {
   @Cacheable()
   getByParamsUnformatted(
     projectParams: Partial<{
-      orgId: string;
-      active: boolean;
+      org_id: string;
```
Comment (Dimple16) [inline src/app/core/services/projects.service.ts:30]: orgID

File: src/app/core/services/projects.service.ts (hunk 8 above)
Comment (Dimple16) [reply src/app/core/services/projects.service.ts:31]: make the same argument changes whereever method is used

File: src/app/core/services/projects.service.ts [hunk 10]
```diff
@@ -94,27 +87,27 @@ export class ProjectsService {
...
     if (typeof orgCategoryIds !== 'undefined' && orgCategoryIds !== null) {
-      params.project_org_category_ids = 'ov.{' + orgCategoryIds.join(',') + '}';
+      params.category_ids = 'ov.{' + orgCategoryIds.join(',') + '}';
     }
   }
 
-  addActiveFilter(active: boolean, params: ProjectParams): void {
-    if (typeof active !== 'undefined' && active !== null) {
-      params.project_active = 'eq.' + active;
+  addActiveFilter(is_enabled: boolean, params: PlatformProjectParams): void {
```
Comment (Dimple16) [inline src/app/core/services/projects.service.ts:108]: `isEnabled`

File: src/app/core/services/projects.service.ts [hunk 11]
```diff
@@ -62,19 +63,11 @@ export class ProjectsService {
...
 
-    return this.apiV2Service
-      .get<ProjectV2, {}>('/projects', {
+    return this.spenderPlatformV1ApiService
+      .get<PlatformApiResponse<PlatformProject[]>>('/projects', {
         params,
       })
-      .pipe(
-        map((res) =>
-          res.data.map((datum) => ({
```
Comment (Dimple16) [inline src/app/core/services/projects.service.ts:71]: verify this with the web app

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/core/services/projects.service.ts (hunk 11 above)
Comment (SahilK-027) [reply src/app/core/services/projects.service.ts:71]: Done 👍

File: src/app/core/services/projects.service.ts (hunk 10 above)
Comment (SahilK-027) [reply src/app/core/services/projects.service.ts:108]: Done 👍

File: src/app/core/services/projects.service.ts (hunk 9 above)
Comment (SahilK-027) [reply src/app/core/services/projects.service.ts:30]: Done 👍

File: src/app/core/services/projects.service.ts (hunk 8 above)
Comment (SahilK-027) [reply src/app/core/services/projects.service.ts:31]: Done 👍

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**